lib/elastic_record/model/joining.rb

```diff
@@ -4,8 +4,19 @@ module Joining
       class JoinChild
         attr_reader :klass, :name, :children, :parent_id_accessor
 
+        def self.build_children(user_input)
+          case user_input
+          when Class, Array
+            Array.wrap(user_input).map { |child| new(klass: child) }
+          when ElasticRecord::Model::Joining::JoinChild
+            Array.wrap(user_input)
+          else
+            raise "Expected a Class, Array, or ElasticRecord::Model::Joining::JoinChild instance as first arg to has_es_children. Received #{user_input} instead."
+          end
```

Comment:
Why stop there?  It can be an Array where some elements are `JoinChild`'s and some are `Class`

Also, there's no need to raise here, because line 20 (the old line 9) will give the right error message.

```suggestion
          Array.wrap(user_input).map do |join_child|
            join_child.is_a?(self) ? join_child : new(klass: join_child)
          end
```